### ISSUE TITLE
SSR with Blaze

### ISSUE THREAD

**Issue by mitar** [kadirahq/flow-router#545]
Blaze is not available on the server side, but [Blaze Components are](https://github.com/peerlibrary/meteor-blaze-components#server-side-rendering). I have not done much testing with SSR, but I am opening ticket here if anyone is interested to try to implement it.

(Just to be clear, if you use Blaze Components, you can also use pure Blaze, so using Blaze Components instead of Blaze gives you also normal Blaze support on the server side. So this can work even for simple Blaze templates.)

See related ticket in Blaze Components repository: https://github.com/peerlibrary/meteor-blaze-components/issues/110

**Comment by arunoda** [comment]
Okay. If so, I think this is super cool.
This is how to add the support for SSR.

Make sure BlazeComponents are rendered inside a FlowRouter action.
In the client, do it normally.
In the server, generate the HTML and send it to FlowRouter like this:

See: https://github.com/kadirahq/react-mounter/blob/master/lib/server.js#L26

**Comment by mitar** [comment]
What about wrapping subscribe calls inside Blaze Components? Will this be done automatically?

**Comment by arunoda** [comment]
Yes. They are automatic. But they need to happen in the same eventloop cycle where FlowRouter .action() is called.

Using it inside a `Tracker.autorun` is fine.

**Comment by mkpazon** [comment]
+1

**Comment by brown2rl** [comment]
I would like to contribute.  Is anyone actively doing this?  Seems like the guide dropped iron router

**Comment by mitar** [comment]
See some discussion [here](https://github.com/meteor/blaze/pull/16) and related tickets.
